google/cloud/internal/async_polling_loop.cc

```diff
@@ -133,23 +133,14 @@ class AsyncPollingLoopImpl
     if (op && op->done()) {
       return promise_.set_value(*std::move(op));
     }
-    // Update the polling policy even on successful requests, so we can stop
-    // after too many polling attempts.
+    // Update the polling policy even on successful requests, so we can
+    // cancel the operation after too many polling attempts.
     if (!polling_policy_->OnFailure(op.status())) {
-      if (op) {
-        // We should not be fabricating a `Status` value here. Rather, we
-        // should cancel the operation and wait for the next poll to return
-        // an accurate status to the user, otherwise they will have no idea
-        // how to react.
-        return promise_.set_value(Status(
-            StatusCode::kDeadlineExceeded,
-            location_ + "() - polling loop terminated by polling policy"));
+      if (!op) {
+        return promise_.set_value(std::move(op).status());
```

Comment:
> After a cancel, the polling operation could fail with a transient and this would be returning with that error, and not the actual status of the long running operation?

Correct, but this is no worse than before; we always returned any transient error after the policy became exhausted.  I added a comment to the effect.

[We could consider also sending the cancel request in the exhausted/poll-failed case.]